satpy/readers/ahi_grid.py

```diff
@@ -0,0 +1,238 @@
+        self.res = AHI_CHANNEL_RES[self.product_name[:3]]
+        if self.areaname == 'fld':
+            self.nlines = AHI_FULLDISK_SIZES[self.res]['y_size']
+            self.ncols = AHI_FULLDISK_SIZES[self.res]['x_size']
+
+        # Set up directory path for the LUTs
+        app_dirs = AppDirs('ahi_gridded_luts', 'satpy', '1.0.1')
+        self.lut_dir = os.path.expanduser(app_dirs.user_data_dir) + '/'
+
+        self.sensor = 'ahi'
+        self.lons = None
+        self.lats = None
+
+    def __del__(self):
+        """Delete the object."""
+        if (self.is_zipped and os.path.exists(self.filename)):
+            os.remove(self.filename)
+
+    def _calibrate(self, data):
+        """Load calibration from LUT and apply."""
+
+        # First, check that the LUT is available. If not, download it.
+        lut_file = self.lut_dir + self.product_name
+        if not os.path.exists(lut_file):
+            self._download_luts()
+        try:
+            # Load file, it has 2 columns: DN + Refl/BT. We only need latter.
+            lut = np.loadtxt(lut_file)[:, 1]
+        except FileNotFoundError:
+            raise FileNotFoundError("No LUT file found:", lut_file)
+
+        # LUT may truncate NaN values, so manually set those in data
+        lut_len = len(lut)
+        data = np.where(data < lut_len - 1, data, np.nan)
+        return lut[data.astype(np.uint16)]
+
+    def _download_luts(self):
+        """LUTs are needed for count->REFL/BT conversion. Download them."""
+        from ftplib import FTP
+        import tempfile
+        import shutil
+        import tarfile
+        import pathlib
+
+        # Check that the LUT directory exists
+        pathlib.Path(self.lut_dir).mkdir(parents=True, exist_ok=True)
+
+        # There is one LUT for each channel
+        flist = ['ext.01', 'vis.01', 'vis.02', 'vis.03',
+                 'sir.01', 'sir.02', 'tir.01', 'tir.02',
+                 'tir.03', 'tir.04', 'tir.05', 'tir.06',
+                 'tir.07', 'tir.08', 'tir.09', 'tir.10']
+
+        # Create a temporary directory for the LUT download
+        tdir = tempfile.gettempdir()
+        fname = tdir + 'tmp.tgz'
+        logger.info("Download AHI LUTs files and store in directory %s",
+                    self.lut_dir)
+
+        # Set up an FTP connection (anonymous) and download
+        ftp = FTP(AHI_REMOTE_LUTS[0])
+        ftp.login('anonymous', 'anonymous')
+        ftp.cwd(AHI_REMOTE_LUTS[1])
+        with open(fname, 'wb') as _fp:
+            ftp.retrbinary("RETR " + AHI_REMOTE_LUTS[2], _fp.write)
+
+        # The file is tarred, here we untar and then remove the downloaded file
+        tar = tarfile.open(fname)
+        tar.extractall(tdir)
+        tar.close()
+        os.remove(fname)
+        # Loop over the LUTs and copy to the correct location
+        for tf in flist:
+            shutil.move(tdir + '/count2tbb/' + tf, self.lut_dir + tf)
+        shutil.rmtree(tdir + '/count2tbb/')
```

Comment:
Please use `os.path.join` for assembling paths, it makes the code more os-independent